scripts/build_shapes.py

```diff
@@ -80,6 +81,48 @@ def download_GADM(country_code, update=False, out_logging=False):
     return GADM_inputfile_gpkg, GADM_filename
 
 
+def restore_country_code_by_name(row, country_code):
+    if row["GID_0"] != country_code:
+        return country_name_2_two_digits(row["COUNTRY"])
+    else:
+        return row["GID_0"]
+
+
+def build_gadm_df(file, layer, cc):
```

Comment:
Maybe there has been a misunderstanding; I was proposing to move to a separate function just the filtering of the GID_0 component.

Basically, keeping the get_GADM_layer as it was and there only adding a line or two like these:

```
if {check config option drop}:
  geodf.drop(list of indices not matching, axis=0, inplace=True);
elif {check config option set to country}:
  geodf["COUNTRY"] = country_code
```

if more lines are needed, we may define a function that contains the rows above and beyond ( e.g. the output of the non-standard zones). By default, the output file shall not be saved
```
def filter_gadm_flag(geodf, config, save_non_standard_geo=False):
   .... stuff
```

Reply:
Mmmm... Agree that the code could be better structured but don't quite get what your idea

Besides, I'm not sure that `drop` is a good default option. Apart of ethical concerns, it'd require some additional changes in the code. (I'll attach a picture to the main PR conversation to explain what is the matter)

So, I suggest to focus in this PR on following GADM conventions with introducing `custom_prescribe` and (probably) `drop` option as the next step

Reply:
I think too that by default shapes shouldn't be dropped, it was just a simple proposal to provide 2 options